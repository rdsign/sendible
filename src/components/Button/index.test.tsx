import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Button } from './index';
import { Template } from '../../../spec/template';

describe('Button', () => {
    const onClick = jest.fn();

    afterEach(() => {
        jest.clearAllMocks();
    });

    const component = (
        <Template>
            <Button
                icon="clock"
                label="Button"
                click={onClick}
                backgroundColor="primary400"
                color="white"
            />
        </Template>
    );

    test('should match snapshot', () => {
        const snapshot = create(component);
        expect(snapshot).toMatchSnapshot();
    });

    test('should call onClick upon click', () => {
        render(component);
        const button = screen.getByRole('button');

        userEvent.click(button);

        expect(onClick).toHaveBeenCalledTimes(1);
    });

    test('if button has a valid icon it should render a svg', () => {
        render(component);

        expect(screen.getByTestId('svg')).toBeInTheDocument();
    });

    test('button text should be "Button"', () => {
        render(component);
        const button = screen.getByRole('button');

        expect(button).toHaveTextContent('Button');
    });
});
