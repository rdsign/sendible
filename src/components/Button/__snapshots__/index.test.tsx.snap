// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #4d36ff;
  border-radius: 5rem;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  padding: 1.4rem 2.6rem;
  -webkit-transition: all 1s cubic-bezier(0.23,1,0.32,1);
  transition: all 1s cubic-bezier(0.23,1,0.32,1);
}

.c0 svg {
  fill: #ffffff;
  height: auto;
  margin-right: 1rem;
  opacity: 0.5;
  width: 1.8rem;
}

.c0:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<button
  className="c0"
  color="white"
  label="Button"
  onClick={[Function]}
  type="button"
>
  <svg
    data-testid="svg"
    fill="inherit"
    height="40"
    viewBox="0 0 40 40"
    width="40"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
      fillRule="evenodd"
    />
  </svg>
  Button
</button>
`;
